lib/vsc/utils/asyncprocess.py

```diff
@@ -179,18 +180,26 @@ def recv_some(p, t=.1, e=False, tr=5, stderr=False, maxread=None):
             len_y += len(r)
         else:
             time.sleep(max((x - time.time()) / tr, 0))
-    return ''.join(y)
+    return b''.join(y)
 
 
 def send_all(p, data):
     """
     Send data to process p
     """
     allsent = 0
+
+    # in Python 3, we must use a bytestring
+    if sys.version_info[0] >= 3:
+        data = data.encode()
+
     while len(data):
         sent = p.send(data)
         if sent is None:
             raise Exception(MESSAGE)
         allsent += sent
-        data = buffer(data, sent)
+        if sys.version_info[0] >= 3:
```

Comment:
`is_py3` function must be part of py23 module; i'm not sure if there an idiom to allow easy switching but this is awful

Reply:
I would argue we should also be aware of potential differences in the various py3 versions. Better to have some server range or some such?

Reply:
added `is_py3` to `vsc.utils.py2vs3`, along with `is_py_ver` which be used to check for `>= 3.6` or so if needed (via `is_py_ver(3, 6)`)